python/ray/data/_internal/simple_block.py

```diff
@@ -97,8 +97,8 @@ def to_pandas(self) -> "pandas.DataFrame":
         return pandas.DataFrame({"value": self._items})
 
     def to_numpy(self, columns: Optional[Union[str, List[str]]] = None) -> np.ndarray:
-        if columns:
-            raise ValueError("`columns` arg is not supported for list block.")
+        if columns is not None:
+            return BlockAccessor.for_block(self.select(columns)).to_numpy()
```

Comment:
`select()` for simple block should be callable, instead of string. wondering why we need to change here?

Reply:
This is to simplify the groupby logic, see the changed in `grouped_dataset.py`: https://github.com/ray-project/ray/pull/30448/files#diff-0e694ceda06b25cb570091a11fc44f31ba8d89ad893f0577946c5b364eaed2cf